specs/chugsplash-manager.md

```diff
@@ -0,0 +1,165 @@
+* Executing an action MUST set the current `_actionIndex` to `true`.
+* The ChugSplashManager MUST emit the event ChugSplashActionExecuted with the active bundle ID, the executor's address, and the action index. The ChugSplashManager MUST announce this event to the registry.
+* If all actions have been executed for the bundle:
+  * The bundle status MUST be set to `COMPLETED`.
+  * The ChugSplashManager MUST increase the `debt` owed to the current executor by the `executorBondAmount`.
+  * The `activeBundleId` MUST be set to `bytes32(0)`.
+  * The ChugSplashManager MUST emit the event ChugSplashBundleCompleted with the active bundle ID, the executor's address, and the action index. The ChugSplashManager MUST announce this event to the registry.
+* Executing an action MUST increase the `totalDebt` and the current executor's `debt` by `block.basefee * gasUsed * (100 + executorPaymentPercentage) / 100)`, where `gasUsed` is calculated using `gasleft()` plus the intrinsic gas (21k) plus the calldata usage.
+
+## Cancelling a ChugSplash bundle
+
+* A ChugSplash bundle MUST only be cancellable by the `admin` of the `ChugSplashManager`.
+* Bundles MUST be in the `APPROVED` state before they can be cancelled.
+* Cancelling a bundle MUST put the bundle in the `CANCELLED` state.
+* Cancelling a bundle MUST remove the active bundle.
+* If an executor has been selected within the last `executionLockTime` seconds (i.e. `block.timestamp` <= `timeClaimed + executionLockTime`):
+  * The ChugSplashManager MUST increase the `debt` owed to the current executor by `BUNDLE_EXECUTION_BOND_AMOUNT`.
+  * The ChugSplashManager MUST increase the `totalDebt` by the `BUNDLE_EXECUTION_BOND_AMOUNT`.
+* The ChugSplashManager MUST emit the event ChugSplashBundleCancelled with the ID of the bundle and the total number of actions executed. The ChugSplashManager MUST announce this event to the registry.
+
+## Withdrawing ETH
+
+* ETH stored in the contract MUST only be withdrawable by the `admin` of the `ChugSplashManager`.
+* The admin ETH withdrawal function MUST NOT be callable while a bundle is in the `APPROVED` state.
+* The admin ETH withdrawal function MUST only allow the admin to withdraw the balance of the account minus `totalDebt`.
+* The ChugSplashManager MUST emit the ETHWithdrawn event with the recipient and the amount. The ChugSplashManager MUST announce this event to the registry.
+
+## Claiming executor payments
+
+* Executors MUST be able to claim payments at any time.
+* Executors MUST NOT be able to claim more payments than their `debt` amount.
+* The ChugSplashManager MUST update the `debt` and `totalDebt` values to reflect the payment.
+* The ChugSplashManager MUST emit the ExecutorPayment event with the recipient and amount. The ChugSplashManager MUST announce this event to the registry.
+
+## Depositing ETH
+
+* Any address should be able to deposit ETH into the `ChugSplashManager` contract at any time.
+
```

Comment:
How do people get paid?  I assume executors aren't doing this for free, and I only see bond repayment mechanisms here.

Reply:
Executor payments are under the section "Claiming executor payments" on line 137, which basically states that executors are able to withdraw the `debt` owed to them. This debt is aggregated in the "Executing a ChugSplash action" section on line 99, which includes an `executorPaymentPercentage` (i.e. the executor's payment).